CodeReview.java

```diff
@@ -38,131 +38,151 @@ public int getLargestNumber() {
 
     public int expensiveFunctionExample() {
         int currNum = 0;
-        if (getLargestNumber() > currNum)
-            currNum = getLargestNumber();
-        for (int i=0; i < getLargestNumber(); ++i)
-            System.out.println(getLargestNumber());
+        int largestNum = getLargestNumber();
+
+        if (largestNum > currNum)
+            currNum = largestNum;
+
+        for (int i = 0; i < largestNum; ++i)
+            System.out.println(largestNum);
 
         return currNum;
     }
 
     public void stringExample() {
-        String oneMillionHello = "";
+        StringBuilder oneMillionHelloSB = new StringBuilder();
         for (int i = 0; i < 1000000; i++) {
-            oneMillionHello = oneMillionHello + "Hello!";
+            oneMillionHelloSB.append("Hello!");
         }
-        System.out.println(oneMillionHello.substring(0, 6));
-
-
-
+        System.out.println(oneMillionHelloSB.toString().substring(0, 6));
     }
 
     public String buildUserQuery() {
         User user = new User();
-        String sql = "Insert Into Users (name, email, pass, address)";
-        sql += " values ('" + user.getFirstname();
-        sql += "', '" + user.getLastname();
-        sql += "', '" + user.getEmail();
-        sql += "', '" + user.getAddress();
-        sql += "')";
-
-        return sql;
+        StringBuilder sbSql
+                = new StringBuilder("Insert Into Users (firstName, lastName, email, address)");
```

Comment:
Vulnerable to SQL Injection - Use PreparedStatements instead. Also extract as Constant.